plugins-root/check_icmp.c

```diff
@@ -456,20 +460,31 @@ main(int argc, char **argv)
 		packets = 5;
 	}
 
-	/* Parse extra opts if any */
-	argv=np_extra_opts(&argc, argv, progname);
-
-	/* support "--help" and "--version" */
-	if(argc == 2) {
-		if(!strcmp(argv[1], "--help"))
-			strcpy(argv[1], "-h");
-		if(!strcmp(argv[1], "--version"))
-			strcpy(argv[1], "-V");
+	/* Parse protocol arguments first */
+	for(i = 1; i < argc; i++) {
+		while((arg = getopt(argc, argv, "vhVw:c:n:p:t:H:s:i:b:I:l:m:64")) != EOF) {
```

Comment:
it feels like we should put the opts string into a variable, otherwise it will soon drift apart.